tests/test.go

```diff
@@ -0,0 +1,217 @@
+	vaults, err := client.Vaults.ListAll(context.Background())
+	if err != nil {
+		panic(err)
+	}
+	for {
+		vault, err := vaults.Next()
+		if errors.Is(err, onepassword.ErrorIteratorDone) {
+			break
+		} else if err != nil {
+			panic(err)
+		}
+
+		fmt.Printf("%s %s\n", vault.ID, vault.Title)
+	}
+	// [developer-docs.sdk.go.list-vaults]-end
+
+	// [developer-docs.sdk.go.list-items]-start
+	items, err := client.Items.ListAll(context.Background(), vaultID)
+	if err != nil {
+		panic(err)
+	}
+	for {
+		item, err := items.Next()
+		if errors.Is(err, onepassword.ErrorIteratorDone) {
+			break
+		} else if err != nil {
+			panic(err)
+		}
+		fmt.Printf("%s %s\n", item.ID, item.Title)
+	}
+	// [developer-docs.sdk.go.list-items]-end
+}
+
+func getAndUpdateItem(client *onepassword.Client, existingVaultID, existingItemID string) {
+	// [developer-docs.sdk.go.update-item]-start
+	// Retrieves the newly created item
+	item, err := client.Items.Get(context.Background(), existingVaultID, existingItemID)
+	if err != nil {
+		panic(err)
+	}
+
+	// Finds the field named "Details" and edits its value
+	for i := range item.Fields {
+		if item.Fields[i].Title == "Details" {
+			item.Fields[i].Value = "updated details"
+		}
+	}
+	item.Title = "New Title"
+	item.Websites = append(item.Websites, onepassword.Website{
+		URL:              "2password.com",
+		Label:            "my second custom website",
+		AutofillBehavior: onepassword.AutofillBehaviorNever,
+	})
+
+	updatedItem, err := client.Items.Put(context.Background(), item)
+	if err != nil {
+		panic(err)
+	}
+	// [developer-docs.sdk.go.update-item]-end
+
+	for _, f := range updatedItem.Fields {
+		if f.Title == "Details" {
+			fmt.Println(f.Value)
+		}
+	}
+}
+
+func resolveSecretReference(client *onepassword.Client) {
+	secretRef := os.Getenv("CORE_SDK_TEST_SECRET_REF")
+
+	// [developer-docs.sdk.go.resolve-secret]-start
+	// Retrieves a secret from 1Password.
+	// Takes a secret reference as input and returns the secret to which it points.
+	secret, err := client.Secrets.Resolve(context.Background(), secretRef)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(secret)
+	// [developer-docs.sdk.go.resolve-secret]-end
+}
+
+func resolveTOTPSecretReference(client *onepassword.Client, vaultID, itemID, fieldID string) {
+	// [developer-docs.sdk.go.resolve-totp-code]-start
+	// Retrieves a TOTP code from 1Password.
+	code, err := client.Secrets.Resolve(context.Background(), fmt.Sprintf("op://%s/%s/%s?attribute=totp", vaultID, itemID, fieldID))
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(code)
+	// [developer-docs.sdk.go.resolve-totp-code]-end
+}
+
+func createAndGetItem(client *onepassword.Client) onepassword.Item {
+	// [developer-docs.sdk.go.create-item]-start
+	vaultID := os.Getenv("CORE_SDK_TEST_VAULT_ID")
+	sectionID := "extraDetails"
+	itemParams := onepassword.ItemCreateParams{
+		Title:    "Login created with the SDK",
+		Category: onepassword.ItemCategoryLogin,
+		VaultID:  vaultID,
```

Comment:
can't this directly be here?
```suggestion
		VaultID:  os.Getenv("CORE_SDK_TEST_VAULT_ID"),
```